var/spack/repos/builtin/packages/variorum/package.py

```diff
@@ -0,0 +1,266 @@
+# Copyright 2013-2019 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+
+import os
+import socket
+
+import llnl.util.tty as tty
+from os import environ as env
+
+
+def cmake_cache_entry(name, value, vtype=None):
+    """
+    Helper that creates CMake cache entry strings used in
+    "host-config" files.
+    """
+    if vtype is None:
+        if value == "ON" or value == "OFF":
+            vtype = "BOOL"
+        else:
+            vtype = "PATH"
+    return 'set({0} "{1}" CACHE {2} "")\n\n'.format(name, value, vtype)
+
+
+class Variorum(Package):
+    """Variorum is a library providing vendor-neutral interfaces for
+       monitoring and controlling underlying hardware features.
+    """
+
+    homepage = "https://variorum.readthedocs.io"
+    url = "https://github.com/LLNL/variorum/archive/v0.1.0-rc1.tar.gz"
+    git = "https://github.com/llnl/variorum.git"
+
+    maintainers = ["brink2"]
+
+    version("dev", branch="dev")
+
+    variant("shared", default=True, description="Build Variorum as shared lib")
+    variant("test", default=True, description="Enable Variorum unit tests")
+
+    # variants for language support
+    variant("python", default=True, description="Build Variorum Python support")
+
+    # variants for dev-tools (docs, etc)
+    variant("docs", default=False, description="Build Variorum's documentation")
+    variant("log", default=False, description="Enable Variorum's logs")
+    variant("debug", default=False, description="Enable Variorum's debug")
+
+    ###########################################################################
+    # package dependencies
+    ###########################################################################
+    depends_on("cmake", type="build")
+    # depends_on("cmake@2.8", type="build")
+    depends_on("hwloc@1.11.9", when="@:0.5.1+hwloc")
+
+    #######################
+    # Documentation related
+    #######################
+    depends_on("py-sphinx", when="+python+docs", type="build")
+
+    #######################
+    # Python
+    #######################
+    # we need a shared version of python b/c linking with static python lib
+    # causes duplicate state issues when running compiled python modules.
```

Comment:
Does this mean that it isn't possible to build `variorum+python~shared`? In this case, you can add a `conflicts`.